test/new-e2e/tests/ndm/snmp/snmp_test.go

```diff
@@ -155,3 +159,36 @@ func (s *snmpDockerSuite) TestSnmp() {
 		assert.Contains(c, metrics, "snmp.sysUpTimeInstance", "metrics %v doesn't contain snmp.sysUpTimeInstance", metrics)
 	}, 5*time.Minute, 10*time.Second)
 }
+
+func (s *snmpDockerSuite) TestSnmpTagsAreStoredOnRestart() {
+	fakeintake := s.Env().FakeIntake.Client()
+	initialMetrics, err := fakeintake.FilterMetrics("snmp.device.reachable",
+		client.WithTags[*aggregator.MetricSeries]([]string{}),
+	)
+	initialTags := initialMetrics[0].Tags
+	_, err = s.Env().RemoteHost.Execute("docker stop dd-snmp")
+	require.NoError(s.T(), err)
+
+	err = fakeintake.FlushServerAndResetAggregators()
+	require.NoError(s.T(), err)
+
+	_, err = s.Env().RemoteHost.Execute(fmt.Sprintf("docker restart %s", s.Env().Agent.ContainerName))
+	require.NoError(s.T(), err)
+
+	err = fakeintake.FlushServerAndResetAggregators()
+	require.NoError(s.T(), err)
+
+	require.EventuallyWithT(s.T(), func(t *assert.CollectT) {
+		metrics, err := fakeintake.FilterMetrics("snmp.device.reachable",
+			client.WithTags[*aggregator.MetricSeries]([]string{}),
+		)
+		require.NoError(t, err)
+		assert.NotEmpty(t, metrics)
+		if len(metrics) == 0 {
+			return
+		}
+		tags := metrics[0].Tags
+		require.ElementsMatch(t, tags, initialTags)
+		require.Len(s.T(), tags, 9)
```

Comment:
These two can be asserts I think.
Also can we check the value of the metric `snmp.device.reachable` to make sure it's 0 (i.e the device is really down, and the tags we have are here because of the cache and not because we fetched them from the running device)?

Reply:
And same here, disregard my comment. We don't want to retry the subtest on such a failure, better to exit early

Reply:
I added a require for the metric https://github.com/DataDog/datadog-agent/pull/26003/commits/69c9d2158aede578535c24fe7ad7c3e9efbfc0b3

Reply:
Just ran into https://github.com/stretchr/testify/pull/1481
It looks like the call to `require` is not supposed to exit the `EventuallyWithT` early, but should only fail the current execution (see Paola's [comment](https://github.com/stretchr/testify/pull/1476#discussion_r1352847918))
We should maybe try to do something like this:
```go
var metrics []Metrics
require.EventuallyWithT(s.T(), func(t *assert.CollectT) {
		metrics, err := fakeintake.FilterMetrics("snmp.device.reachable",
			client.WithTags[*aggregator.MetricSeries]([]string{}),
		)
		require.NoError(t, err)
		assert.NotEmpty(t, metrics)
	}, 5*time.Minute, 5*time.Second)
}
tags := metrics[0].Tags
require.ElementsMatch(t, tags, initialTags)
require.Len(s.T(), tags, 9)
```
to ensure we fail without retrying for 5 mins

Reply:
applied your suggestion https://github.com/DataDog/datadog-agent/pull/26003/commits/2fd8dfd6aecdd24c2494b31bca3413fda7066156